qiskit/primitives/base/validation.py

```diff
@@ -0,0 +1,231 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Primitive validation methods.
+
+Note that these are not intended to be part of the public API of base primitives
+but are here for backwards compatibility with deprecated functions.
+"""
+
+from __future__ import annotations
+
+from collections.abc import Sequence
+import typing
+import numpy as np
+
+from qiskit.circuit import QuantumCircuit, ControlFlowOp, Measure
+from qiskit.quantum_info.operators import SparsePauliOp
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+from ..utils import init_observable
+
+if typing.TYPE_CHECKING:
+    from qiskit.opflow import PauliSumOp
+
+
+def validate_estimator_args(
```

Comment:
If `validate_estimator_args` and `validate_sampler_args` are to be deprecated in the near future, it may not be necessary to make it public in the first place.

Reply:
I agree to make the validation functions private.